src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/HandlerUtils.java

```diff
@@ -66,4 +66,11 @@ public static String getAfterHearingReqSuffix() {
     public static boolean isNabaEnabled(AsylumCase asylumCase) {
         return (asylumCase.read(IS_NABA_ENABLED, YesOrNo.class)).orElse(YesOrNo.NO) == YesOrNo.YES;
     }
+
+    //Updated method to check if it is a rep journey
+    public static boolean isLegalRepJourney(AsylumCase asylumCase) {
+        return asylumCase.read(JOURNEY_TYPE, JourneyType.class)
+                .map(journeyType -> journeyType == JourneyType.REP)
```

Comment:
Are you sure we are updating the field - JOURNEY_TYPE. I think it gets written in AIP journey for sure, but can you check? Otherwise - suggestion: Check if LegalRep details are present on caseData.

Reply:
Fixed